src/pages/index.js

```diff
@@ -58,48 +27,175 @@ const editModalDescriptionInput = editModal.querySelector(
+    "Content-Type": "application/json",
+  },
+});
 
-const plusImage = document.getElementById("image-plus");
-plusImage.src = plusSrc;
+api
+  .getAppInfo()
+  .then(([userInfo, cards]) => {
+    profileName.textContent = userInfo.name;
+    profileDescription.textContent = userInfo.about;
+    profileAvatar.src = userInfo.avatar;
+    cards.forEach((item) => {
+      renderCard(item, "append");
+    });
+  })
+  .catch(console.error);
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch((err) => {
+      console.error(err);
+      submitBtn.disabled = false;
+    })
+    .finally(() => {
+      setButtonText(submitBtn, false, "Save", "Saving...");
+    });
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  renderCard(inputValues);
-  cardForm.reset();
-  disableButton(cardSubmitButton, settings);
-  closeModal(cardModal);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Save", "Saving...");
+  submitBtn.disabled = true;
+
+  api
+    .newCardInfo({ name: cardNameInput.value, link: cardLinkInput.value })
+    .then((newCard) => {
+      renderCard(newCard, "prepend");
+      closeModal(cardModal);
+      cardForm.reset();
+
+      disableButton(submitBtn, settings);
+    })
+    .catch((err) => {
+      console.error(err);
+      submitBtn.disabled = false;
+    })
+    .finally(() => {
+      setButtonText(submitBtn, false, "Save", "Saving...");
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Save", "Saving...");
+  submitBtn.disabled = true;
+
+  api
+    .editAvatarInfo({ avatar: avatarInput.value })
+    .then((data) => {
+      avatarEl.src = data.avatar;
+      closeModal(avatarModal);
+      avatarForm.reset();
+
+      disableButton(submitBtn, settings);
+    })
+    .catch((err) => {
+      console.error(err);
+      submitBtn.disabled = false;
+    })
+    .finally(() => {
+      setButtonText(submitBtn, false, "Save", "Saving...");
+    });
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Delete", "Deleting...");
+  submitBtn.disabled = true;
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard;
+      selectedCard.remove();
+      closeModal(deleteModal);
+      selectedCard = cardElement;
```

Comment:
* When I delete cards, there is an error https://snipboard.io/0CcKG4.jpg 

`selectedCard = cardElement;` is not needed here


You should set `undefined` here to return the initial value